cli/command/cli.go

```diff
@@ -187,6 +187,36 @@ func (cli *DockerCli) BuildKitEnabled() (bool, error) {
 	return cli.ServerInfo().OSType != "windows", nil
 }
 
+// HooksEnabled returns whether plugin hooks are enabled.
+func (cli *DockerCli) HooksEnabled() bool {
+	// legacy support DOCKER_CLI_HINTS env var
+	if v := os.Getenv("DOCKER_CLI_HINTS"); v != "" {
+		enabled, err := strconv.ParseBool(v)
+		if err != nil {
+			return false
+		}
+		return enabled
+	}
+	// use DOCKER_CLI_HOOKS env var value if set and not empty
+	if v := os.Getenv("DOCKER_CLI_HOOKS"); v != "" {
```

Comment:
@thaJeztah same argument for the env var name as you mentioned [here](https://github.com/docker/cli/pull/4960)?

Reply:
no, different one – this is user facing, used to override/enable/disable hooks if the user wants.

Reply:
Are we sure? Hints are user facing, not hooks (even though they are the same thing now, i still think the naming could be problematic)

Reply:
> Are we sure?

I am, but maybe I'm missing something. Can you tell me why you believe this env var to enable/disable hooks shouldn't be user facing?

Reply:
Because plugins themselves aren't __really__ user facing, that's my worry here. Also, I can imagine that a user might want to disable hints without disabling other "hooks" functionality (admittedly, we don't have other functionality right now, but even just the name of the settings I need to change being HOOKS over HINTS would be confusing to me as a user)

Reply:
So for the user to be aware of hooks, they should also be aware of the cli plugins, or else we're just adding to the confusion IMHO

Reply:
Users who would be aware of this env var would also be aware of hooks. Also, clearly we want to advertise a way to turn this off, to offer an easy escape hatch if we accidentally bork things with hooks. This is very different from an env var that a user would "never" touch, such as the one we were discussing with the sockets.

Reply:
> So for the user to be aware of hooks, they should also be aware of the cli plugins, or else we're just adding to the confusion IMHO

Also, some users are. We sometimes tell users to download x binary of Compose and place it in the `~/.docker/cli-plugins` folder if they quickly need a new version. Most users aren't but also this is a public fact about how the docker CLI works. There's separate packages in debian repos, etc.

Reply:
> Also, I can imagine that a user might want to disable hints without disabling other "hooks" functionality

Also, this is [supported](https://github.com/docker/cli/pull/4376#issuecomment-2008580277). Would be a matter of removing the `"hooks"` line for a particular plugin.

Reply:
> Users who would be aware of this env var would also be aware of hooks. Also, clearly we want to advertise a way to turn this off, to offer an easy escape hatch if we accidentally bork things with hooks. This is very different from an env var that a user would "never" touch, such as the one we were discussing with the sockets.

Fair, so maybe in the future we could add an env var specific to turning off HINTS if we end up adding more functionality to hooks in general (the annoyance to the user was what i was imagining the env var was for, not the scenario of hooks borking stuff)

Reply:
> Also, this is https://github.com/docker/cli/pull/4376#issuecomment-2008580277. Would be a matter of removing the "hooks" line for a particular plugin.

Argh true..I still have to wrap my head around all these configs, as i'm not yet totally convinced of using env vars for some things, the config.json for others, etc especially when the features are so closely related. I'll resolve this thread

Reply:
> (the annoyance to the user was what i was imagining the env var was for, not the scenario of hooks borking stuff)

This isn't even a huge point because it'll ship off by default, and the general "off/on" switch will be the `features` map in `config.json`
```json
{
  "auths": {},
  "features": {
    "hooks": "true",
  },
  "plugins": {
    "hints": {
      "hooks": "pull,build"
    }
  }
}
```

the env var just provides the same ability, but it can be more useful to set in some environments/it's also just good practice to provide env vars to override config for these things.

Reply:
But yeah, I think the rest of that we can consider in the future when we have more substantial things to figure out and hooks are doing more things.

Reply:
> as i'm not yet totally convinced of using env vars for some things, the config.json for others, etc especially when the features are so closely related.

I'm happy to continue this conversation elsewhere though, but there shouldn't be confusion here – there is only one env var being introduced `DOCKER_CLI_HOOKS`, which is a general override regardless of what is configured.

`DOCKER_CLI_HINTS` was carried over from the previous implementation, and does the same thing ("big override button"). Other than that (those are big overrides, not configuration), things are configured in the `config.json`.